File: src/classes/edge.py
'''Classe responsável por representar o conteudo de um Edge '''


class Edge:
    def __init__(self, source_id: str, destination_id: str | None, weight, is_pedagio = False):
        self.source_id = source_id
        self.destination_id = destination_id
        self.weight = weight
        self.is_pedagio = is_pedagio

    def to_string(self):
        destination = self.destination_id
        source = self.source_id
        weight = self.weight

        return str(f"({source}) is connected to {destination} with weight {weight}")

File: Backend/pathfinder/demo/src/main/java/com/example/pathfinder/graph/Edge.java
package com.example.pathfinder.graph;

import java.util.Map;
import java.util.HashMap;

public class Edge {
    private String sourceId;
    private String destinationId;
    private int weight;

    public String getSourceId() {
        return sourceId;
    }

    public String getDestinationId() {
        return destinationId;
    }

    public int getWeight() {
        return weight;
    }

    public Edge(String sourceId, String destinationId, int weight) {
        this.sourceId = sourceId;
        this.destinationId = destinationId;
        this.weight = weight;
    }

    public String toString() {
        String destination = this.destinationId;
        String source = this.sourceId;
        int weight = this.weight;

        return String.format("(%s) is connected to %s with weight %d", source, destination, weight);
    }

    public java.util.Map<String, Object> toMap() {
        Map<String, Object> edgeMap = new HashMap<>();
        edgeMap.put("destination", this.destinationId);
        edgeMap.put("source", this.sourceId);
        edgeMap.put("weight", this.weight);

        return edgeMap;
    }
}